plugins/SocketPlugin.js

```diff
@@ -98,51 +119,34 @@ function SocketPlugin() {
           // Extract possible data to send
           var data = null;
           if (contentLength !== undefined) {
-            data = this.sendBuffer.slice(this.sendBufferIndex - contentLength, this.sendBufferIndex);
+            data = this.sendBuffer.slice(
+                this.sendBufferIndex - contentLength, this.sendBufferIndex);
           }
 
           var httpRequest = new XMLHttpRequest();
-          httpRequest.open(httpMethod, url);
+          httpRequest.open(httpMethod, this._getURL(targetURL, httpMethod));
           if (contentType !== undefined) {
             httpRequest.setRequestHeader('Content-type', contentType);
           }
           httpRequest.responseType = "arraybuffer";
 
           var thisHandle = this;
           httpRequest.onload = function (oEvent) {
-            var content = this.response;
-            if (content) {
-              // Recreate header
-              var header = new TextEncoder('utf-8').encode(
-                'HTTP/1.0 ' + this.status + ' ' + this.statusText + '\r\n' +
-                this.getAllResponseHeaders() + '\r\n');
-              // Concat header and content
-              thisHandle.response = new Uint8Array(header.byteLength + content.byteLength);
-              thisHandle.response.set(header, 0);
-              thisHandle.response.set(new Uint8Array(content), header.byteLength);
-              thisHandle.responseLength = thisHandle.response.byteLength;
-
-              that.primHandler.signalSemaphoreWithIndex(thisHandle.readSemaIndex);
-            } else {
-              thisHandle.status = that.Socket_OtherEndClosed;
-            }
+            thisHandle._handleResponse(this);
           };
 
           httpRequest.onerror = function(e) {
+            var url = thisHandle._getURL(targetURL, httpMethod, true);
```

Comment:
What if we started with the proxy (because of needsProxy), and arrive here with an error? Shouldn't we then retry *without* the proxy?